runtime/CMakeLists.txt

```diff
@@ -0,0 +1,361 @@
+	target_compile_definitions(j9vm_compiler_defines INTERFACE TR_TARGET_POWER TR_HOST_POWER)
+endif() #TODO OTHER PLATFORMS
+
+#TODO should use findCuda
+include_directories( /usr/local/cuda-5.5/include )
+
+
+#set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/lib")
+
+add_library(j9vm_interface INTERFACE)
+
+target_include_directories(j9vm_interface
+	INTERFACE
+		oti
+		include
+		omr/include_core
+		nls
+)
+
+#TODO deprecated
+add_library(j9vm_nls INTERFACE)
+target_include_directories(j9vm_nls 
+	INTERFACE
+
+)
+
+
+add_library(j9vm_gc_includes INTERFACE)
+target_include_directories(j9vm_gc_includes
+	INTERFACE
+		$<TARGET_PROPERTY:omrgc,INTERFACE_INCLUDE_DIRECTORIES>
+		${j9vm_SOURCE_DIR}/gc_include
+		${j9vm_SOURCE_DIR}/gc_glue_java
+		${j9vm_SOURCE_DIR}/gc_base
+		${j9vm_SOURCE_DIR}/gc_stats
+		${j9vm_SOURCE_DIR}/gc_structs
+)
+
+#setup OMR library overrides
+set(OMR_GC_LIB "j9gc" CACHE STRING "Name of the GC library to use")
+set(OMR_HOOK_LIB "j9hookable" CACHE STRING "Name of the hook library to link against")
+set(OMR_PORT_LIB "j9prt" CACHE STRING "Name of the port library to link against")
+set(OMR_THREAD_LIB "j9thr" CACHE STRING "Name of the thread library to link against")
+set(OMR_TRACE_LIB "j9trc" CACHE STRING "Name of the trace library to link against")
+
+add_subdirectory(gc_glue_java)
+add_subdirectory(omr)
+
+set_target_properties(omrsig PROPERTIES LINKER_LANGUAGE C)
+
+# Do out hookgen stuff
+add_hookgen(oti/j9jit.hdf ${CMAKE_CURRENT_SOURCE_DIR}/oti/jithook.h ${CMAKE_CURRENT_SOURCE_DIR}/oti/jithook_internal.h)
+add_hookgen(oti/j9vm.hdf ${CMAKE_CURRENT_SOURCE_DIR}/oti/vmhook.h ${CMAKE_CURRENT_SOURCE_DIR}/oti/vmhook_internal.h)
+add_hookgen(oti/zipCachePool.hdf ${CMAKE_CURRENT_SOURCE_DIR}/oti/vmzipcachehook.h ${CMAKE_CURRENT_SOURCE_DIR}/oti/vmzipcachehook_internal.h)
+
+add_dependencies(j9vm_interface
+		${CMAKE_CURRENT_SOURCE_DIR}/oti/jithook.h
+		${CMAKE_CURRENT_SOURCE_DIR}/oti/vmhook.h
+		${CMAKE_CURRENT_SOURCE_DIR}/oti/vmzipcachehook.h
+)
+
+add_custom_target(j9vm_hookgen
+	DEPENDS
+	${CMAKE_CURRENT_SOURCE_DIR}/oti/jithook.h
+	${CMAKE_CURRENT_SOURCE_DIR}/oti/vmhook.h
+	${CMAKE_CURRENT_SOURCE_DIR}/oti/vmzipcachehook.h
+)
+if(TARGET omrglue)
+	set(J9VM_LEGACY_GLUE 1)
+else()
+	set(J9VM_LEGACY_GLUE 0)
+endif()
+
+set(J9VM_OLD_SHARED_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
+set(CMAKE_SHARED_LIBRARY_SUFFIX "29${CMAKE_SHARED_LIBRARY_SUFFIX}")
+
+if(J9VM_MODULE_BCUTIL)
+	add_subdirectory(bcutil)
+endif()
+if(J9VM_MODULE_BCVERIFY)
+	add_subdirectory(bcverify)
+endif()
+if(J9VM_MODULE_CASSUME)
+	add_subdirectory(cassume)
+endif()
+if(J9VM_MODULE_CFDUMPER)
+	add_subdirectory(cfdumper)
+endif()
+add_subdirectory(clear_vmi)
+if(J9VM_MODULE_CODERT_VM)
+	add_subdirectory(codert_vm)
+endif()
+add_subdirectory(cuda)
+if(J9VM_MODULE_DBGEXT)
+	add_subdirectory(dbgext)
+endif()
+#add_subdirectory(ddr_plugin)
+#add_subdirectory(ddrext)
+if(OMR_HOST_OS STREQUAL "win")
```

Comment:
Basically because I haven't implemented them. I just went through and put `add_subdirectory` for all the directories and went through un-commenting them one by one when i brought them online